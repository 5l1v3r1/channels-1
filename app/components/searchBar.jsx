// @flow
import React from 'react';
import { IconSearch, IconCancel } from './icons';
import styles from './searchBar.scss';

const SearchBar = ({
  disabled,
  placeholder,
  value
}: {
  disabled: boolean,
  placeholder: string,
  value: string
}) => (
  <form className={styles.searchBar}>
    <IconSearch styles={styles} />
    <input
      className={styles.searchInput}
      disabled={disabled}
      placeholder={placeholder}
      value={value}
    />
    <IconCancel
      styles={styles}
      value={value}
    />
  </form>
);

export default SearchBar;
