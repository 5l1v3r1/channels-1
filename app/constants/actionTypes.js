// @flow
const actionTypes = {
  REQUEST_ACCESS_TOKEN: 'REQUEST_ACCESS_TOKEN',
  RECEIVE_ACCESS_TOKEN: 'RECEIVE_ACCESS_TOKEN',
  ACCESS_TOKEN_ERROR: 'ACCESS_TOKEN_ERROR',
  REQUEST_CHANNELS: 'REQUEST_CHANNELS',
  RECEIVE_CHANNELS: 'RECEIVE_CHANNELS',
  CHANNELS_ERROR: 'CHANNELS_ERROR',
  TOGGLE_CHANNEL_LIST: 'TOGGLE_CHANNEL_LIST',
  UPDATE_ACTIVE_CHANNEL: 'UPDATE_ACTIVE_CHANNEL',
  UPDATE_STATUS: 'UPDATE_STATUS',
  ENABLE_SEARCH_BAR: 'ENABLE_SEARCH_BAR'
};

export default actionTypes;
