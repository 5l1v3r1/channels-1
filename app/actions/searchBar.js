// @flow
import actionTypes from '../constants/actionTypes';
import type { Action } from '../constants/typeAliases';

export function enableSearchBar(title: string): Action {
  return {
    type: actionTypes.ENABLE_SEARCH_BAR,
    payload: title
  };
}
