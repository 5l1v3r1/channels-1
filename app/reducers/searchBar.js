// @flow
import actionTypes from '../constants/actionTypes';
import type { Action, SearchBarState } from '../constants/typeAliases';

const initialState = {
  disabled: true,
  placeholder: 'Search Channels',
  value: ''
};

export default function searchBar(state: SearchBarState = initialState, { type, payload }: Action): SearchBarState {
  switch (type) {
    case actionTypes.ENABLE_SEARCH_BAR:
      return {
        ...state,
        disabled: false,
        placeholder: !payload ? 'Search' : `Search ${payload}`
      };
    default:
      return state;
  }
}
