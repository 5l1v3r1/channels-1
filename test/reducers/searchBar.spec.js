import searchBar from '../../app/reducers/searchBar';
import actionTypes from '../../app/constants/actionTypes';

describe('searchBar reducer', () => {
  it('should return initial state', () => {
    expect(searchBar(undefined, {})).toMatchSnapshot();
  });

  it('should handle ENABLE_SEARCH_BAR', () => {
    const action = {
      type: actionTypes.ENABLE_SEARCH_BAR,
      payload: 'DevTips'
    };
    expect(searchBar(undefined, action)).toMatchSnapshot();
  });

  it('should handle unknown action type', () => {
    const action = { type: 'unknown' };
    expect(searchBar(undefined, action)).toMatchSnapshot();
  });
});
