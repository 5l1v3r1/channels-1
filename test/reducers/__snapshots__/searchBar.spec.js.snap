// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchBar reducer should handle ENABLE_SEARCH_BAR 1`] = `
Object {
  "disabled": false,
  "placeholder": "Search DevTips",
  "value": "",
}
`;

exports[`searchBar reducer should handle unknown action type 1`] = `
Object {
  "disabled": true,
  "placeholder": "Search Channels",
  "value": "",
}
`;

exports[`searchBar reducer should return initial state 1`] = `
Object {
  "disabled": true,
  "placeholder": "Search Channels",
  "value": "",
}
`;
